### PR TITLE
Add global roles data to course roles API

### PR DESCRIPTION
**Pull request by tehreem-sadat**
Related Issue: https://github.com/nelc/futurex-openedx-extensions/issues/69#issue-2521426679

### CODE REVIEW

File: futurex_openedx_extensions/dashboard/serializers.py [hunk 1]
```diff
@@ -506,6 +511,15 @@ def get_tenants(self, obj: get_user_model) -> Any:
...
         return self.roles_data.get(obj.id, {}) if self.roles_data else {}
 
+    def get_global_roles(self, obj: get_user_model) -> Any:  # pylint:disable=no-self-use
+        """Return the global roles."""
+        global_roles = []
+        roles_dict = get_user_course_access_roles(obj)['roles']
+        for role, details in roles_dict.items():
+            if details.get('global_role', False):
+                global_roles.append(role)
+        return global_roles
```
Comment (shadinaif) [inline futurex_openedx_extensions/dashboard/serializers.py:521]: this works great @tehreem-sadat , thank you!

now, I'll drift you away from this for-loop. I added the existing `global_role` flag before realizing that global roles can only be global. I mean, we can know if the role is global or not by its name!

Can you please remove `'global_role'` completely from `get_user_course_access_roles` and refactor `get_global_roles` accordingly?

**Review by shadinaif — CHANGES_REQUESTED** [review]

File: futurex_openedx_extensions/dashboard/serializers.py (hunk 1 above)
Comment (tehreem-sadat) [reply futurex_openedx_extensions/dashboard/serializers.py]: @shadinaif Changes have been made PR is ready for re-review.

File: futurex_openedx_extensions/dashboard/serializers.py [hunk 2]
```diff
@@ -506,6 +515,15 @@ def get_tenants(self, obj: get_user_model) -> Any:
...
         return self.roles_data.get(obj.id, {}) if self.roles_data else {}
 
+    def get_global_roles(self, obj: get_user_model) -> Any:  # pylint:disable=no-self-use
+        """Return the global roles."""
+        global_roles = []
+        roles_dict = get_user_course_access_roles(obj)['roles']
+        for role in roles_dict:
+            if role in COURSE_ACCESS_ROLES_GLOBAL:
+                global_roles.append(role)
+        return global_roles
```
Comment (shadinaif) [inline futurex_openedx_extensions/dashboard/serializers.py:525]: how about
```suggestion
        roles_dict = get_user_course_access_roles(obj)['roles']
        return [role for role in roles_dict if role in COURSE_ACCESS_ROLES_GLOBAL]
```

**Review by shadinaif — APPROVED** [review]
Thank you @tehreem-sadat , one small change and it's good to go

File: futurex_openedx_extensions/dashboard/serializers.py (hunk 1 above)
Comment (shadinaif) [reply futurex_openedx_extensions/dashboard/serializers.py]: thank you!

**Pull request merged**